tests/bias/test_qdm_bias_correction.py

```diff
@@ -0,0 +1,298 @@
+"""pytests QDM bias correction calculations"""
+
+import os
+import shutil
+
+import h5py
+import numpy as np
+import pytest
+import xarray as xr
+
+from sup3r import TEST_DATA_DIR
+from sup3r.bias.bias_calc import QuantileDeltaMappingCorrection
+from sup3r.bias.bias_transforms import local_qdm_bc
+from sup3r.preprocessing.data_handling import DataHandlerNC
+
+FP_NSRDB = os.path.join(TEST_DATA_DIR, "test_nsrdb_co_2018.h5")
+FP_CC = os.path.join(TEST_DATA_DIR, "rsds_test.nc")
+FP_CC_LAT_LON = DataHandlerNC(FP_CC, "rsds").lat_lon
+
+with xr.open_dataset(FP_CC) as fh:
+    MIN_LAT = np.min(fh.lat.values.astype(np.float32))
+    MIN_LON = np.min(fh.lon.values.astype(np.float32)) - 360
+    TARGET = (float(MIN_LAT), float(MIN_LON))
+    SHAPE = (len(fh.lat.values), len(fh.lon.values))
+
+
+@pytest.fixture(scope="session")
```

Comment:
Fixtures are fine but i have a few opinions: 1) I think they rerun for every function they're used in so they don't really speed anything up, 2) if you want to run the .py as a script you'll get an error; they only work with pytest... so 3) i generally prefer just normal python functions that are executed in test functions because they serve the same purpose but are more simple. 

Definitely open to other opinions on this but they generally seem like syntactic sugar.

Reply:
@grantbuster Is syntactic sugar bad? Google search AI tells me "Syntactic sugar is a term used in computer science to describe syntax changes that make code easier to read or write." Sounds positive to me lol. Also, I don't think I've ever wanted to run test .py files without prepending with pytest. What would be an example where this would be useful?

Reply:
Haha fair point. IMO it can be good but it's often language or package specific so requires googling to understand what's going on (increases cognitive load? barrier for new developers?). 

Sometimes I find it easier to debug tests in a python script instead of a pytest function. You can run python scripts in ipython or jupyter but only add breakpoints in pytests. 

This is all a bit of personal bias. We use pytest fixtures all over the place and they definitely serve a purpose. I just wanted to call out some tradeoffs i've seen and why i've gone away from using them.

Reply:
My understanding is that the `scope` defines how to re-use, or not, the fixture. The default is indeed per function, but we can re-use it. Check [the scope explanation](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#scope-sharing-fixtures-across-classes-modules-packages-or-session). I just checked and the time difference is not too much, but it adds up once we run a full test.

`pytest tests/bias/test_qdm_bias_correction.py --durations=10`

For debugging, it might help running it like: `pytest tests/bias/test_qdm_bias_correction.py --pdb`

Reply:
Let me split this in two questions so we can make a decision on how to move forward:
- Do you agree that `scope="session"` or even `scope="module"` allow re-using the fixture, thus should speed up the tests?
- Does `pytest --pdb ...` is a fair option to debug new tests?

Reply:
Just checking. @grantbuster and @bnb32 have an opinion about using `@pytest.fixture`? I suggest using it as it is, but I have no problem in working without it if you think so.

Reply:
I'm fine with keeping as is unless Grant has objections.

Reply:
Totally fine with using the fixtures. I appreciate your note on the scope.